third_party/libev/ev.h

```diff
@@ -696,16 +696,21 @@ EV_API_DECL void ev_resume  (EV_P) EV_NOEXCEPT;
 
 /* these may evaluate ev multiple times, and the other arguments at most once */
 /* either use ev_init + ev_TYPE_set, or the ev_TYPE_init macro, below, to first initialise a watcher */
-#define ev_init(ev,cb_) do {			\
-  ((ev_watcher *)(void *)(ev))->active  =	\
-  ((ev_watcher *)(void *)(ev))->pending = 0;	\
-  ev_set_priority ((ev), 0);			\
-  ev_set_cb ((ev), cb_);			\
+#define ev_init(ev,cb_) do { 							\
+  memset ((char *)(ev) + offsetof(ev_watcher, active), 0, sizeof(int)); 	\
+  memset ((char *)(ev) + offsetof(ev_watcher, pending), 0, sizeof(int));	\
+  ev_set_priority ((ev), 0);							\
+  ev_set_cb ((ev), cb_);							\
 } while (0)
 
 #define ev_io_modify(ev,events_)             do { (ev)->events = (ev)->events & EV__IOFDSET | (events_); } while (0)
 #define ev_io_set(ev,fd_,events_)            do { (ev)->fd = (fd_); (ev)->events = (events_) | EV__IOFDSET; } while (0)
-#define ev_timer_set(ev,after_,repeat_)      do { ((ev_watcher_time *)(ev))->at = (after_); (ev)->repeat = (repeat_); } while (0)
+#define ev_timer_set(ev,at_,repeat_) do {       								\
+  ev_tstamp __ev_timer_set_at = (at_);          								\
+  memcpy ((char *)(ev) + offsetof(ev_watcher_time, at), &__ev_timer_set_at, sizeof(__ev_timer_set_at));		\
+  ev_tstamp __ev_timer_set_repeat = (repeat_);  								\
+  memcpy ((char *)(ev) + offsetof(ev_timer, repeat), &__ev_timer_set_repeat, sizeof(__ev_timer_set_repeat));	\
```

Comment:
I don't think that we should patch libev like this - it will complicate updating to a new version. Better live without strict aliasing IMO.

BTW checkpatch complains about mixed tabs and spaces in this patch.

Reply:
Discussed f2f: compiler warnings from libunwind sources should be ignored — dropped these changes.